src/ServiceProvider.php

```diff
@@ -58,39 +70,46 @@ private function registerComponents()
             return $factory;
         });
 
-        $this->app->singleton(Firebase\Auth::class, static function (Application $app) {
+        $this->app->singleton(Firebase\Auth::class, static function ($app) {
+            /** @var LaravelApplication|LumenApplication $app */
             return $app->make(Firebase\Factory::class)->createAuth();
         });
         $this->app->alias(Firebase\Auth::class, 'firebase.auth');
 
-        $this->app->singleton(Firebase\Database::class, static function (Application $app) {
+        $this->app->singleton(Firebase\Database::class, static function ($app) {
+            /** @var LaravelApplication|LumenApplication $app */
             return $app->make(Firebase\Factory::class)->createDatabase();
         });
         $this->app->alias(Firebase\Database::class, 'firebase.database');
 
-        $this->app->singleton(Firebase\DynamicLinks::class, static function (Application $app) {
+        $this->app->singleton(Firebase\DynamicLinks::class, static function ($app) {
+            /** @var LaravelApplication|LumenApplication $app */
             $defaultDynamicLinksDomain = $app->make('config')['firebase']['dynamic_links']['default_domain'] ?? null;
 
             return $app->make(Firebase\Factory::class)->createDynamicLinksService($defaultDynamicLinksDomain);
         });
         $this->app->alias(Firebase\DynamicLinks::class, 'firebase.dynamic_links');
 
-        $this->app->singleton(Firebase\Firestore::class, static function (Application $app) {
+        $this->app->singleton(Firebase\Firestore::class, static function ($app) {
+            /** @var LaravelApplication|LumenApplication $app */
             return $app->make(Firebase\Factory::class)->createFirestore();
         });
         $this->app->alias(Firebase\Firestore::class, 'firebase.firestore');
 
-        $this->app->singleton(Firebase\Messaging::class, static function (Application $app) {
+        $this->app->singleton(Firebase\Messaging::class, static function ($app) {
+            /** @var LaravelApplication|LumenApplication $app */
```

Comment:
You can actually just typehint `Illuminate\Contracts\Container\Container` since both Laravel and Lumen Application extends it.

Reply:
Thanks for the tip 👍, I like this much better 🥰